lib/cc/analyzer/container.rb

```diff
@@ -0,0 +1,70 @@
+module CC
+  module Analyzer
+    class Container
+      TIMEOUT = 15 * 60 # 15m
+
+      def initialize(image, command = nil, log = nil)
+        @image = image
+        @command = command
+        @log = log || NullContainerLog.new
+
+        @output_delimeter = "\n"
+        @on_output = ->(*)
+
+        @timed_out = false
+        @stderr_io = StringIO.new
+      end
+
+      def on_output(delimeter = "\n", &block)
+        @output_delimeter = delimeter
+        @on_output = block
+      end
+
+      def run(options)
+        @log.started(@image)
+
+        pid, _, out, err = POSIX::Spawn.popen4(*docker_run_command)
```

Comment:
I forgot to implement `docker_run_command` but it'll be `docker run` + default options + `options`